WordPress/gutenberg
# Move the DisableNonContentBlocks component to the editor package

Related #52632 

## What?

Since the site editor is now using the "mode" prop from the EditorProvider to indicates how to render the page/template, it should be part of the responsibility of the EditorProvider to lock/unlock blocks depending on this mode. This PR does just that by moving the `DisableNonContentBlocks` to the `EditorProvider`.

Ultimately, this will allow us to actually use the same template/editor modes in the site and post editors.

## Testing Instructions

1- Just make sure that the site editor for editing pages continue to work as expected: templates locked by default and you can switch...
